robottelo/libvirt_discovery.py

```diff
@@ -77,16 +78,22 @@ def __init__(
             self.image_dir = image_dir
         if mac is None:
             self.mac = _gen_mac_for_libvirt()
-        if bridge is None:
-            self.bridge = settings.vlan_networking.bridge
+        if network is None:
+            if settings.vlan_networking.bridge or settings.vlan_networking.network:
```

Comment:
If `network` is None (which is default), and neither of settings are defined (which shouldn't happen, but technically can), then we will thrown AttributeError on line 92, when trying to check if `self.network` is falsy. That's probably not what you wanted.

These three nested ifs make me think that there's probably better way to write this. Maybe helper function/method? Something that would take `network` and `network_type` as arguments, and return `self.network_type`  and `self.network` values back? Untested proposal below:

```python
def get_network_settings(network, network_type):
    if network is not None:
        return (network, network type)
    if settings.vlan_networking.bridge:
        return ('bridge', settings.vlan_networking.bridge)
    if settings.vlan_networking.network:
        return ('network', settings.vlan_networking.network)
    return (None, None)  # return None tuple, so tuple unpacking and assignment always works
                         # we bail out in caller if `self.network` is falsy
```

Reply:
I'm counting on [settings.verify()](https://github.com/SatelliteQE/robottelo/blob/master/robottelo/config/base.py#L1148), which accepts exactly 1 of bridge or network in `[vlan_networking]`